pkg/healthcheck/healthcheck.go

```diff
@@ -1582,6 +1595,32 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil && len(apiStatus.Status.Conditions) == 0 {
+		return err
+	}
+
+	available := false
+	var errorMessage string
```

Comment:
I think we should have a default errorMessage that will be used if no conditions of type "Available" are found.  As it is written, this would use the nil string as the error message if the conditions list is empty, for example.